ocaml/xapi/db_gc_util.ml

```diff
@@ -222,6 +222,23 @@ let gc_host_metrics ~__context =
         Db.Host_metrics.destroy ~__context ~self:hmetric)
     all_host_metrics
 
+let gc_certificates ~__context =
+  let all_certificates = Db.Certificate.get_all_records ~__context in
+  let get_certificates =
+    List.map (fun self -> Db.Host.get_certificates ~__context ~self)
+  in
+  let host_certificates =
+    get_certificates (Db.Host.get_all ~__context) |> List.flatten
+  in
+  List.iter
+    (fun (cert, record) ->
+      if
```

Comment:
I would use `List.filter` followed by `List.iter` - it's not more efficient but I think it fits the paradigm and is readable.

Reply:
I think a comment about the intention would be also good: which certs you want to remote

Reply:
not (A || B) == (not A) && (not B) - would the  latter logic be more readable?

Reply:
I did the && logic before, not sure which one is less confusing. I'll add a comment to state we want to remove the certificates that have a host associates (`host and `internal_host) with an invalid host reference.